watertap/unit_models/tests/test_gac.py

```diff
@@ -367,7 +364,96 @@ def test_solution_robust(self, gac_frame_robust):
         assert pytest.approx(12830000, rel=1e-3) == value(m.fs.unit.elap_time)
         assert pytest.approx(10.68, rel=1e-3) == value(m.fs.unit.bed_area)
 
-    @pytest.mark.requires_idaes_solver
+    @pytest.mark.component
+    def test_reporting_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+        m.fs.unit.report()
+
+    @pytest.mark.component
+    def test_variable_sensitivity_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+
+        import idaes.core.util.scaling as iscale
+        from pyomo.common.collections import ComponentSet
+
+        def var_sens_generator(
```

Comment:
This looks like something that would need its own testing. I also think it would benefit from more comments to make it easier to follow. We could think about adding a utility under `water.core.util.scaling`, then bring the function in here, but as mentioned, you'd also need a test file for the new scaling function.

Reply:
There are niches with this single use case that may need redefined if adding to `water.core.util.scaling`, say testing the handling vol/mass/mole flow and other compatibilities. I think if this does get migrated and more work done testing and commenting are sure to be included.

Reply:
It does strike me as odd that you are defining this method inside a test. I would have thought that this is either something more generally useful that would warrant a utility function, or something that can be achieved in other ways.

Reply:
I think @hunterbarber intended to get eyes on this function before taking the next step of adding it as a utility function. So we should first make sure we know everything that's happing in the function and make a determination on whether it should be added in the first place. Thus, more comments describing what is happening throughout the function would be helpful in proceeding.